frontend/src/data/MetricQuery.ts

```diff
@@ -48,6 +48,10 @@ export class MetricQueryResponse {
       this.error = input as Error;
       this.data = [];
       this.invalidValues = {};
+    } else if (input.length <= 0) {
```

Comment:
This comment isn't anchored properly because github doesn't allow me to comment on the right line...

Should we rename `shouldShowError` to `shouldShowDataWarning`?

Reply:
I've got it as shouldShowMissingDataMessage now to match the message content but we can do shouldShowDataWarning if you prefer.

Reply:
What you have works too!